src/jdk.compiler/share/classes/com/sun/tools/javac/processing/PrintingProcessor.java

```diff
@@ -524,13 +522,29 @@ private void printFormalTypeParameters(Parameterizable e,
             List<? extends TypeParameterElement> typeParams = e.getTypeParameters();
             if (!typeParams.isEmpty()) {
                 writer.print(typeParams.stream()
-                             .map(tpe -> annotationsToString(tpe) + tpe.toString())
+                             .map(tpe -> annotationsToString(tpe) + tpe.toString() + printTypeVariableBoundsIfNeeded(tpe))
                              .collect(Collectors.joining(", ", "<", ">")));
                 if (pad)
                     writer.print(" ");
             }
         }
 
+        private String printTypeVariableBoundsIfNeeded(TypeParameterElement tpe) {
+            List<? extends TypeMirror> printableBounds =
+                    tpe.getBounds()
+                       .stream()
+                       .filter(type -> !isUnimportantObjectType(type))
```

Comment:
Perhaps the changes would be easier to ready if the method as "isImportantObjectType()"?

Reply:
Thanks, done under:
https://github.com/openjdk/jdk/pull/25002/commits/e78e88a6a3f0967d12f3f19a989f5e8e033eb724